rosidl_typesupport_introspection_tests/include/rosidl_typesupport_introspection_tests/gtest/message_introspection_test.hpp

```diff
@@ -76,12 +76,14 @@ class MessageIntrospectionTest : public SharedLibraryTest
   MakeTypeErasedMessage() const
   {
     std::allocator<MessageT> allocator;
-    std::function<void(void *)> type_erased_message_deleter = [ = ](void * ptr) mutable {
-        finalize_message(ptr, message_descriptor_);
+    auto message_descriptor = GetMessageDescriptor();
+    std::function<void(void *)> type_erased_message_deleter =
+      [allocator, message_descriptor](void * ptr) mutable {
```

Comment:
I'm certainly not an expert on C++20, but is there some reason we can't do something like:

```
    std::function<void(void *)> type_erased_message_deleter =
      [this, allocator](void * ptr) mutable {
```

?  `message_descriptor_` is part of the class, so capturing `this` should be enough, no?

Reply:
Yep, for some reason my brain assumed a similar warning would be thrown, so I went full on explicit declaration, but that works out, and it's a lot cleaner. Sold! Thanks for the review :)